mm2src/coins/lightning.rs

```diff
@@ -186,20 +222,19 @@ impl LightningCoin {
             status: HTLCStatus::Pending,
             created_at: (now_ms() / 1000) as i64,
             last_updated: (now_ms() / 1000) as i64,
-        })
+        };
+        self.db.add_or_update_payment_in_db(payment_info.clone()).await?;
+        Ok(payment_info)
     }
 
-    async fn keysend(
+    pub(crate) async fn keysend(
         &self,
         destination: PublicKey,
         amount_msat: u64,
         final_cltv_expiry_delta: u32,
-    ) -> SendPaymentResult<DBPaymentInfo> {
+    ) -> Result<PaymentInfo, PaymentError> {
```

Comment:
`MmError` is used to track the error path. In the current implementation, we'll get the only place where `keysend` was used, but it's not definitely bad.
I actually think it can be overhead in some cases, but I can't formulate my own rules when I prefer to use or not to use.
> So are we supposed to use MmError whenever possible?

Probably, when a crate or a module doesn't depend on `mm2_err_handle` crate.